internal/infraestructure/github-connection/github-connection.go

```diff
@@ -2,54 +2,185 @@ package github_connection
 
 import (
 	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
 	"github.com/google/go-github/v61/github"
 )
 
-type GithubConnectionParams struct {
-	RepositoryName    string
-	Token             string
-	RepoOwner         string
-	PullRequestNumber int
+type GithubConnection struct {
+	Client            *github.Client `json:"client"`
+	RepositoryName    string         `json:"repository_name"`
+	RepoOwner         string         `json:"repo_owner,omitempty"`
+	PullRequestNumber int            `json:"pull_request_number"`
+	EventName         string         `json:"event_name"`
 }
 
-func NewGithubConnection(params GithubConnectionParams) (*GithubConnection, error) {
+type FileChange struct {
+	github.CommitFile
+	Position int `json:"position"`
+}
+
+func NewGithubConnection() (*GithubConnection, error) {
+	token := os.Getenv("GITHUB_TOKEN")
+	if token == "" {
+		return nil, fmt.Errorf("GITHUB_TOKEN environment variable is not set")
+	}
+
+	repoFullName := os.Getenv("GITHUB_REPOSITORY")
+	if repoFullName == "" {
+		return nil, fmt.Errorf("GITHUB_REPOSITORY environment variable is not set")
+	}
+	repoParts := strings.Split(repoFullName, "/")
+	if len(repoParts) != 2 {
+		return nil, fmt.Errorf("invalid GITHUB_REPOSITORY format: %s", repoFullName)
+	}
+	repoOwner := repoParts[0]
+	githubRepositoryName := repoParts[1]
+
+	pullRequestNumberStr := os.Getenv("GITHUB_PR_NUMBER")
+	if pullRequestNumberStr == "" {
+		return nil, fmt.Errorf("GITHUB_PR_NUMBER environment variable is not set")
+	}
+	pullRequestNumber, err := strconv.Atoi(pullRequestNumberStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid GITHUB_PR_NUMBER: %w", err)
+	}
 
-	GithubClient := github.NewClient(nil).WithAuthToken(params.Token)
+	eventName := os.Getenv("GITHUB_EVENT_NAME")
+	if eventName == "" {
+		return nil, fmt.Errorf("GITHUB_EVENT_NAME environment variable is not set")
+	}
+	GithubClient := github.NewClient(nil).WithAuthToken(token)
 
```

Comment:
Code Review 2024-08-18 02:28:52 - New lines 43 to 47:

	eventName := os.Getenv("GITHUB_EVENT_NAME")
	if eventName == "" {
		return nil, fmt.Errorf("GITHUB_EVENT_NAME environment variable is not set")
	}
	GithubClient := github.NewClient(nil).WithAuthToken(token)